subsys/debug/Kconfig

```diff
@@ -228,6 +228,23 @@ config EARLY_CONSOLE
 	  initialize its driver earlier than normal, in order to get the stdout
 	  sent through the console at the earliest stage possible.
 
+config EARLY_PRINT
```

Comment:
We already have CONFIG_EARLY_CONSOLE, why can't this be used? This looks very custom to a use case and we need to generalize this

Reply:
The `CONFIG_EARLY_CONSOLE` focuses on `input` and `output` at a early boot stage but cannot be earlier than interrupt controller init. The `CONFIG_EARLY_CONSOLE` is initialized after z_cstart(). However, when developers want to bring up a new platform, they probably need to print something (only output) for debugging even before interrupt controller or MMU. I found x86 also has this kind of early print https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/x86/core/early_serial.c), we need a similar one for Arm64, that's the main reason I want to introduce this but this is more generic.

Reply:
> I found x86 also has this kind of early print [`main`/arch/x86/core/early_serial.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/x86/core/early_serial.c?rgh-link-date=2023-11-29T02%3A07%3A35Z)), we need a similar one for Arm64, that's the main reason I want to introduce this but this is more generic.

You are right, I was actually thinking about this one which is more lile what you are looking for and it is controlled with CONFIG_X86_VERY_EARLY_CONSOLE which pretty much is the same as EARLY_PRINT. 

We need to generalize this, it does not make sense to have different architecture have different ways for doing the same thing, I do not think any of this supports input at this stage, even if they have CONSOLE in the name.

Reply:
Okay Anas, if you prefer, I could change the `X86_VERY_EARLY_CONSOLE`, and make it under the `EARLY_PRINT`?

Reply:
wait on that, let me think about it a bit more, @jhedberg  any input on this?

Reply:
Sure, take your time

Reply:
Yeah, at first sight it looks like the `X86_VERY_EARLY_CONSOLE` functionality should be moved behind this new more generic Kconfig option.

Reply:
Can I ask if there are any more suggestions?